packages/rxtranscribe/src/operators/bufferBetweenSilence.js

```diff
@@ -0,0 +1,112 @@
+import {of, iif, generate, EMPTY, forkJoin, BehaviorSubject} from 'rxjs';
+import {finalize, zip, takeLast, share, mergeMap, map, tap, pairwise, reduce, skipWhile, expand, takeWhile, takeUntil, take} from 'rxjs/operators';
+import toVAD, {VadEvents} from './toVAD';
+
+const MIN_CHUNK_SIZE = 10000;
+const VAD_SAMPLE_SIZE = 1000;
+
+/**
+ *
+ * @param vadOptions options that vad uses for the operator
+ * @param _toVAD The VAD operator
+ */
+const bufferBetweenSilence = function bufferBetweenSilence({
+  vadOptions = { sampleRate: 16000 },
+  _toVAD = toVAD,
+}) {
+  return chunk$ => {
+    let offset = 0;
+
+    // Replays the same chunk that was supplied
+    const chunkSub$ = chunk$.pipe(share());
+
+    const pump = (subject$) => ([vadRes, chunk]) => {
+      console.log('Offset', offset);
+      console.log('Chunk Length', chunk.length);
+      // console.log('Pump values', [vadRes, chunk]);
+      offset += VAD_SAMPLE_SIZE;
+      console.log('this should be false', chunk.length - offset < MIN_CHUNK_SIZE);
+      console.log('this should also be false', vadRes === VadEvents.SILENCE);
+      if (chunk.length - offset < MIN_CHUNK_SIZE || vadRes === VadEvents.SILENCE) {
+        console.log('We should not be getting here.');
+        return subject$.complete();
+      }
+      console.log('We should definitely get here.');
+      return subject$.next(offset);
+    };
+
+    const byteOffset$ = new BehaviorSubject(0);
+    const pumpNext = pump(byteOffset$);
+
+    // Creates slices for VAD to run against
+    const slicer$ = byteOffset$.pipe(
+      // tap(r => console.log('Inside the byteOffset', offset)),
+      zip(chunkSub$),
+      // tap(r => console.log('Ran zip on chunk sub', r)),
+      map(([newOffset, chunk]) => (
+        chunk.subarray(
+          chunk.length - newOffset - VAD_SAMPLE_SIZE - 1, chunk.length - newOffset - 1
+        )
+      )),
+      // tap(r => console.log('Got the new offset', r)),
+      _toVAD({...vadOptions}),
+      // tap(r => console.log('Vad Result', r)),
+      map(([,{type}]) => type),
+      zip(chunkSub$),
+      tap(pumpNext),
+      // reduce((acc, val) => val)
+    );
```

Comment:
I can see why this is tricky...  I'm having a hard time getting my head around what the `slicer$` observable does...  I think what it's intended to do is take each fileChunk and iterate over it until it finds silence...

Reply:
What I don't understand is this:  Is there a relationship between fileChunks in the source observable?  Like, why not just take each fileChunk and run a simple `map` over it?

Reply:
The issue I'm having is waiting for the _ToVAD operator to finish. If its run as an observable then I need a recursive observable to keep running until the offset is found. If I run it as a promise, then I need to keep waiting for the response again and again inside of an async function, but there is some combination of things happening that it doesn't wait for the vad call as a promise to finish and as an observable it doesn't loop.